src/unix.js

```diff
@@ -61,6 +70,46 @@ function escapeArgBash(arg, { interpolation, quoted }) {
   return result;
 }
 
+/**
+ * Escapes a shell argument for use in csh.
+ *
+ * @param {string} arg The argument to escape.
+ * @param {object} options The escape options.
+ * @param {boolean} options.interpolation Is interpolation enabled.
+ * @param {boolean} options.quoted Is `arg` being quoted.
+ * @returns {string} The escaped argument.
+ */
+function escapeArgCsh(arg, { interpolation, quoted }) {
+  let result = arg
+    .replace(/[\0\u0008\u001B\u009B]/gu, "")
+    .replace(/\r?\n|\r/gu, " ");
+
+  if (interpolation) {
+    result = result
+      .replace(/\\/gu, "\\\\")
+      .replace(/(^|\s)(~)/gu, "$1\\$2")
+      .replace(/(["#$&'()*;<>?[`{|])/gu, "\\$1")
+      .replace(/([\t ])/gu, "\\$1");
+
+    result = result
+      .split("")
+      .map(
+        // For an unknown reason when a character whose utf-8 encoding includes
+        // the No-Break Space (0xA0) appears *after* an escaped character, the
+        // C shell will hang. This is mitigated by explicitly escaping all such
+        // characters.
```

Comment:
It appears I'm not the first to run into this, there's an open bug report for this problem at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=995013